src/packagedcode/rubygems.py

```diff
@@ -648,92 +628,172 @@ def normalize(gem_data, known_fields=known_fields):
+            gemspec_data['authors'] = authors
+            continue
+
+        # update the value of email
+        if '.email ' in individual:
+            email = individual.split('= ')
+            email = email[1].strip('[]%q{}').replace('"', '')
+            gemspec_data['email'] = email
+            continue
+
+        # update the value of summary
+        if '.summary ' in individual:
+            summary = individual.split('= ')
+            summary = summary[1].strip('\'"%q{}')
+            gemspec_data['summary'] = summary
+            continue
+
+        # update the value of description
+        if '.description ' in individual:
+            description = individual.split('= ')
+            description = description[1].strip('\'"%q{}')
+            gemspec_data['description'] = description
+            continue
+
+        # update the value of homepage
+        if '.homepage ' in individual and 'homepage_uri' not in individual:
+            homepage = individual.split('= ')
+            homepage = homepage[1].strip('\'"%q{}')
+            gemspec_data['homepage'] = homepage
+            continue
 
-class GemSpec(object):
+        # update the value of license
+        if '.license ' in individual:
+            license = individual.split('= ')
+            license = license[1].strip('\'"%q{}')
+            gemspec_data['license'] = license
+            continue
+
+        # update the value of dependencies
+        if 'dependency' in  individual:
+            dep_version = individual
+            # >>> s = 's.add_runtime_dependency(%q<mechanize>, [">= 0"])'
+            # >>> z = re.split(", ", s)
+            # >>> z = z[1].strip("\"'[]()")
+            # >>> z
+            # '>= 0'
+            dep_version = re.split(", ", dep_version)
+            try:
+                dep_version = dep_version[1].strip(" \"'[]()")
+            except IndexError:
+                dep_version = None
+
+            # >>> s = 's.add_runtime_dependency(%q<mechanize>, [">= 0"])'
+            # >>> d = re.split("'|<|>", s)
+            # >>> d
+            # ['s.add_runtime_dependency(%q', 'mechanize', ', ["', '= 0"])']
+            dependency = re.split("'|\"|<|>", individual)
+            dependencies[dependency[1]] = dep_version
+            continue
+
+    # stores dependencies list into dictionary
+    if not dependencies:
+        dependencies = None
+    gemspec_data['dependencies'] = dependencies
+
+    return gemspec_data
+
+
+def build_packages_from_gemspec(gemspec_data):
     """
-    Represent a Gem specification.
+    Return a package built from gemspec.
     """
+    name = gemspec_data.get('name')
+    version = gemspec_data.get('version')
+    description = gemspec_data.get('description')
+    homepage_url = gemspec_data.get('homepage')
+    package = RubyGem(
+        name=name,
+        version=version,
+        description=description,
+        homepage_url=homepage_url
```

Comment:
Its slightly confusing to create a minimal package at first and keep adding to it. Why not create a mapping dict and just call `package = RubyGem(**mapped_data)` at the end?

Reply:
@majurg I tried `package = RubyGem(**mapped_data)` but not able to implement it. 
I saw https://github.com/nexB/scancode-toolkit/pull/2062/files implementation, in this you have implemented by for loop. 
I think whatever is implemented in this is good and is similar to other packaged code also. :)